### ISSUE TITLE
Inconsistent logging of properties and attachments on caught and unhandled crashes

### ISSUE THREAD

**Issue by justintoth** [microsoft/appcenter-sdk-dotnet#815]
### **Description**

There is inconsistent behavior when logging "caught" exceptions in a try/catch block with Crashes.TrackError and logging "uncaught" exceptions (which happens automatically by the AppCenter SDK.)

Issue #1: When calling Crashes.TrackError I can set custom properties to attach to the error report, such as device name, user name, where they are in the app, etc... However, there is no way to set these properties on an unhandled exception that is logged internally on app restart by the AppCenter SDK.

Issue #2: When logging an unhandled exception on app restart, I can add an attachment (e.g. a screenshot of the erroring screen)  to the error report. However, when calling Crashes.TrackError there is no option to attach a screenshot.

So basically, I'm looking for parity between Crashes.TrackError and the internal logging of uncaught exceptions, so that we can add properties and attachments to BOTH types of crashes. Otherwise, our crashes will be very inconsistent, as some will have properties, some will have attachments, but never both.

### **Repro Steps**

Not needed, the above should suffice I think.

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - Xamarin.Android
2. Which SDK version are you using?
    - e.g. 1.13.0
4. Which OS version did you experience the issue on?
    - Android 8.0
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Google Nexus 6P Genymotion Emulator

**Comment by winnieli1208** [comment]
Hi @justintoth,

Thank you for opening an issue. We currently don't support adding attachments for handled errors (crashes), only unhandled errors, and don't support custom properties for unhandled errors (crashes), only handled errors. 

We understand this isn't an ideal experience and have an item in our backlog to improve feature parity between unhandled errors and handled errors. Unfortunately, due to other priorities right now, I don't have a concrete timeline but will update you as soon as I have more details. 

Thank you!

**Comment by Thijmen** [comment]
Looking forward to this!

**Comment by chrisaswain** [comment]
I have also noticed that when using TrackError, I don't see the "Stack traces" in AppCenter.

**Comment by vvechkanov** [comment]
Hi @chrisaswain!

Thank you for contacting us. To properly help you, I need to better understand the problem. Do you mean that stack trace is empty for an error, that you log in? 

I think it would be great if you will create a new "Problem report", for this problem. When you will create it, there are questions, about the information we need to know to help you. Like a version of SDK, platform, and so on. It also will be more convenient for all of us to track "TrackError issue" separately from this feature request.

**Comment by DmitriyKirakosyan** [comment]
Hi @justintoth ,
You can send attachments with `TrackError` now. Please see [Handled errors doc](https://docs.microsoft.com/en-us/appcenter/sdk/crashes/xamarin#handled-errors).
I'm closing this issue but feel free to reopen if you have additional questions.
